File: src/content/product.ts
import { z, defineCollection } from "astro:content";

const productCollection = defineCollection({
  type: "content",
  schema: z.object({
    code: z.string(),
    name: z.string(),
    description: z.string(),
    defaultPrice: z.number(),
    discountedPrice: z.number().optional(),
    currency: z.string().optional(),
    image: z.string().optional(),
    moq: z.number().optional(),
    tags: z.array(z.string()).optional(),
    stock: z.number().optional(),
    availability: z.string().optional(),
    weight: z.number().optional(),
    dimensions: z.string().optional(),
    SKU: z.string().optional(),
    manufacturer: z.string().optional(),
    warranty: z.string().optional(),

    SEOTitle: z.string().optional(),
    SEODescription: z.string().optional(),
    SEOKeywords: z.array(z.string()).optional(),
  }),
});

export default productCollection;

File: cli/models/product/product.go
package products

type Product struct {
	Code            string   `yaml:"code"`
	Name            string   `yaml:"name"`
	Description     string   `yaml:"description"`
	DefaultPrice    float64  `yaml:"defaultPrice"`
	DiscountedPrice *float64 `yaml:"discountedPrice,omitempty"`
	Currency        *string  `yaml:"currency,omitempty"`
	Image           *string  `yaml:"image,omitempty"`
	Moq             *int     `yaml:"moq,omitempty"`
	Tags            []string `yaml:"tags,omitempty"`
	Stock           *int     `yaml:"stock,omitempty"`
	Availability    *string  `yaml:"availability,omitempty"`
	Weight          *float64 `yaml:"weight,omitempty"`
	Dimensions      *string  `yaml:"dimensions,omitempty"`
	SKU             *string  `yaml:"SKU,omitempty"`
	Manufacturer    *string  `yaml:"manufacturer,omitempty"`
	Warranty        *string  `yaml:"warranty,omitempty"`
	SEOTitle        *string  `yaml:"SEOTitle,omitempty"`
	SEODescription  *string  `yaml:"SEODescription,omitempty"`
	SEOKeywords     []string `yaml:"SEOKeywords,omitempty"`
}
